### ISSUE TITLE
missing Uniform instances for matrices

### ISSUE THREAD

**Issue by ekmett** [haskell-opengl/OpenGL#68]
There doesn't appear to be a way to set up mat4 matrices for e.g. perspective matrices in shaders.

**Comment by bananu7** [comment]
Just for reference for people who don't want to use `GLUtil` - the way it did that was rather simple:

```
unUL :: GL.UniformLocation -> GLint
unUL = unsafeCoerce

uniformGLMat4 :: GL.UniformLocation -> GL.SettableStateVar (GL.GLmatrix GL.GLfloat)
uniformGLMat4 loc = GL.makeSettableStateVar aux
  where aux m = GL.withMatrix m $ \_ -> GLRaw.glUniformMatrix4fv loc' 1 1
        loc' = unUL loc
```

Apart from the ugly `unsafeCoerce` to pass the uniform location to the native code, it's no rocket science. What are the plans to solve that? Are you perhaps waiting for a PR?

**Comment by svenpanne** [comment]
This has been fixed in b1b50031f7880b918fc6ec056342c5c89f118213, I'm about to make a new release on Hackage.
